### PR TITLE
Feat: CLI for InApp boilerplate creation

### PR DESCRIPTION
**Pull request by RomainBillot**
# Intro
This PR is huge because we embed applications code into the CLI. You can safely ignore all changes in `/cli/boilerplate` and focus on the rest, which is not that huge.

This PR bring a new feature, a CLI for creating InApp boilerplate application. This should be as easy as one command : `inapp create -n my-inapp -t front`.

The InApp name (`-n`) can be in any case, it is normalized in kebab-case anyway.

The InApp type (`-t`) can be one of: front, back or full. Note that is you've selected front only, you will only have client application code, and the deployment should only deploy a client application (build VueJs app and upload it to an S3).

General idea: The boilerplate is conditional based on the InApp type, we'll use it for any future custom InApp development.

There is a README for this repo in this branch that should be easy to read, please notify anything weird / not understandable.

# CLI stack
The CLI is written in Golang and embed all application code (boilerplate). It used cobra and viper under the hood for CLI / configuration management. There is a Github action `.github/workflows` that build the CLI application and distribute it as binaries for common architecture (windows64/32, linux64/32 and darwin64/32).

# Testing locally
You can follow the README (on the `feat/cli` branch) and generate a new InApp, if it's not easy, let us know 😉 .

# Before merging this PR
We should create a new tag and update the README to reflect new CLI binaries version.

We are open to any comment, please, be hard on us 😈

### CODE REVIEW

**Comment by AmelieChevalier** [comment]
@RomainBillot Attention tu as fait ta PR vers master je sais pas si c'est voulu ou si c'était pour develop!

**Comment by RomainBillot** [comment]
> @RomainBillot Attention tu as fait ta PR vers master je sais pas si c'est voulu ou si c'était pour develop! (j'ai pas encore tout regardé par ailleurs)

Merci de ton attention, en effet 👍

**Comment by RomainBillot** [comment]
@martinmaurice @goniti @AmelieChevalier 
Est-ce que vous pourrez essayer la CLI de votre côté s'il vous plaît, pour que j'ai vos retours tant sur le code que sur l'utilisation de la CLI ? Thanks !

`curl -SL https://github.com/360medics/inapp-template/releases/download/0.0.4/inapp-0.0.4-darwin-amd64.tar.gz | tar -zxC /usr/local/bin`

**Comment by RomainBillot** [comment]
Development automation -> deployment automation dans le README

File: cli/boilerplate/stacks/client/src/shims-vue.d.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+/* eslint-disable */
+declare module '*.vue' {
+	import type { DefineComponent } from 'vue';
+	const component: DefineComponent<{}, {}, any>;
+	export default component;
+}
```
Comment (Pascal-Legroux) [inline cli/boilerplate/stacks/client/src/shims-vue.d.ts:6]: indent looks strange

Comment (RomainBillot) [reply cli/boilerplate/stacks/client/src/shims-vue.d.ts:6]: resolve with .editorconfig, thanks !

File: cli/boilerplate/stacks/api/src/app.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+// Healthcheck route for load balancer
+app.get('/', (req, res) => {
+  res.send({ message: 'API is healthy!' });
+});
+
+// global middlewares registration
+app.use(cors());
+app.use(helmet());
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({ extended: false }));
```
[the remark this answers is not in the document]
Comment (RomainBillot) [reply cli/boilerplate/stacks/api/src/app.ts:20]: Thanks for the tips ! Done

File: cli/boilerplate/stacks/client/src/components/PDFViewer.vue [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    <div v-if="pdfLink" class="pdfviewer">
+        <a
+            v-if="!isMobile()"
+            class="link"
+            :href="`https://f.360medics.com/pdfviewer/index.html?f=${pdfLink}`"
+            target="_blank"
+        >
+            {{ pdfName }}
+        </a>
+        <a v-else class="link" :href="'medics://viewer?m_source=https://f.360medics.com/pdfviewer/index.html?f=' + pdfLink">
```
[the remark this answers is not in the document]
Comment (RomainBillot) [reply cli/boilerplate/stacks/client/src/components/PDFViewer.vue:11]: Totally agreed.

We may need to use a formatter to do such things.
@AmelieChevalier any ideas on this ?

**Pull request merged**